src/test/java/org/folio/circulation/support/utils/DateFormatUtilTests.java

```diff
@@ -0,0 +1,251 @@
+  }
+
+  @Test
+  void shouldFormatZonedDateTimeOptional() {
+    final String result = DateFormatUtil.formatDateTimeOptional((ZonedDateTime) null);
+
+    assertNull(result);
+  }
+
+  @Test
+  void shouldFormatOffsetDateTimeOptional() {
+    final String result = DateFormatUtil.formatDateTimeOptional((OffsetDateTime) null);
+
+    assertNull(result);
+  }
+
+  @Test
+  void shouldFormatJodaDateTimeOptional() {
+    final String result = DateFormatUtil.formatDateTimeOptional((DateTime) null);
+
+    assertNull(result);
+  }
+
+  @Test
+  void shouldFormatZonedDateTime() {
+    final ZonedDateTime date = ZonedDateTime.parse(FORMATTED_DATE_TIME);
+    final String result = DateFormatUtil.formatDateTime(date);
+
+    assertEquals(FORMATTED_DATE_TIME, result);
+  }
+
+  @Test
+  void shouldFormatOffsetDateTime() {
+    final OffsetDateTime date = OffsetDateTime.parse(FORMATTED_DATE_TIME);
+    final String result = DateFormatUtil.formatDateTime(date);
+
+    assertEquals(FORMATTED_DATE_TIME, result);
+  }
+
+  @Test
+  void shouldFormatDateAsDateTime() {
+    final LocalDate date = LocalDate.parse(FORMATTED_DATE);
+    final String result = DateFormatUtil.formatDateTime(date);
+
+    assertEquals(FORMATTED_DATE_TIME_FROM_DATE, result);
+  }
+
+  @Test
+  void shouldFormatJodaDateTime() {
+    final DateTime date = DateTime.parse(FORMATTED_DATE_TIME);
+    final String result = DateFormatUtil.formatDateTime(date);
+
+    assertEquals(FORMATTED_DATE_TIME, result);
+  }
+
+  @Test
+  void shouldParseZonedDateTime() {
+    final ZonedDateTime date = ZonedDateTime.parse(FORMATTED_DATE_TIME);
+    final ZonedDateTime result = DateFormatUtil.parseDateTime(FORMATTED_DATE_TIME);
+
+    assertEquals(date, result);
+  }
+
+  @ParameterizedTest
+  @CsvSource(value = {
+    "0, null, null, " + FORMATTED_DATE_TIME_NOW,
+    "1, Z, null, " + FORMATTED_DATE_TIME_NOW,
+    "2, null, " + FORMATTED_DATE_TIME + ", " + FORMATTED_DATE_TIME,
+    "3, Z, " + FORMATTED_DATE_TIME + ", " + FORMATTED_DATE_TIME
+  }, nullValues = {"null"})
+  void shouldParseZonedDateTimeWithZone(int id, ZoneId zone, String value, String match) {
+    ClockUtil.setClock(Clock.fixed(Instant.parse(FORMATTED_DATE_TIME_NOW), ZoneOffset.UTC));
+
+    final ZonedDateTime date = ZonedDateTime.parse(match);
+    final ZonedDateTime result = DateFormatUtil.parseDateTime(value, zone);
+
+    assertEquals(date, result, "For test " + id);
+  }
+
+  @ParameterizedTest
+  @CsvSource(value = {
+    "0, null, null, null",
+    "1, UTC, null, null",
+    "2, null, " + FORMATTED_DATE_TIME + ", " + FORMATTED_DATE_TIME,
+    "3, UTC, " + FORMATTED_DATE_TIME + ", " + FORMATTED_DATE_TIME
+  }, nullValues = {"null"})
+  void shouldParseJodaDateTimeOptional(int id, ZoneId zone, String value, String match) {
+    ClockUtil.setClock(Clock.fixed(Instant.parse(FORMATTED_DATE_TIME_NOW), ZoneOffset.UTC));
+
+    final ZonedDateTime date = match == null
+      ? null
+      : ZonedDateTime.parse(match);
```

Comment:
I think that branching in tests reduces the readability of the tests and makes them harder to maintain.

I understand that you disagree and so these will likely not be changed.